ingester2/src/persist/handle.rs

```diff
@@ -128,6 +128,7 @@ use crate::{
 pub(crate) struct PersistHandle {
     /// Task handles for the worker tasks, aborted on drop of all
     /// [`PersistHandle`] instances.
+    #[allow(dead_code)]
     worker_tasks: Vec<AbortOnDrop<()>>,
```

Comment:
```suggestion
    _worker_tasks: Vec<AbortOnDrop<()>>,
```

it's not dead-per-say - it has a meaning! By retaining the handles, they're not aborted, because the `drop` impl of `AbortOnDrop` kills the workers (making it panic safe).